Sources/FronteggSwift/services/CredentialManager.swift

```diff
@@ -38,7 +38,8 @@ public class CredentialManager {
                 kSecClass: kSecClassGenericPassword,
                 kSecAttrService: serviceKey ?? "frontegg",
                 kSecAttrAccount: key,
-                kSecValueData: valueData
+                kSecValueData: valueData,
+                kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlock
```

Comment:
Why did you decide kSecAttrAccessibleAfterFirstUnlock and not kSecAttrAccessible

Reply:
[kSecAttrAccessible
](https://developer.apple.com/documentation/security/ksecattraccessible)
Has constants:
@constant kSecAttrAccessibleWhenUnlocked Item data can only be accessed
        while the device is unlocked. This is recommended for items that only
        need be accesible while the application is in the foreground.  Items
        with this attribute will migrate to a new device when using encrypted
        backups.
    @constant kSecAttrAccessibleAfterFirstUnlock Item data can only be
        accessed once the device has been unlocked after a restart.  **This is
        recommended for items that need to be accesible by background
        applications.** Items with this attribute will migrate to a new device
        when using encrypted backups.
    @constant kSecAttrAccessibleAlways Item data can always be accessed
        regardless of the lock state of the device.  This is not recommended
        for anything except system use. Items with this attribute will migrate
        to a new device when using encrypted backups.
    @constant kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly Item data can
        only be accessed while the device is unlocked. This is recommended for
        items that only need to be accessible while the application is in the
        foreground and requires a passcode to be set on the device. Items with
        this attribute will never migrate to a new device, so after a backup
        is restored to a new device, these items will be missing. This
        attribute will not be available on devices without a passcode. Disabling
        the device passcode will cause all previously protected items to
        be deleted.
    @constant kSecAttrAccessibleWhenUnlockedThisDeviceOnly Item data can only
        be accessed while the device is unlocked. This is recommended for items
        that only need be accesible while the application is in the foreground.
        Items with this attribute will never migrate to a new device, so after
        a backup is restored to a new device, these items will be missing.
    @constant kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly Item data can
        only be accessed once the device has been unlocked after a restart.
        This is recommended for items that need to be accessible by background
        applications. Items with this attribute will never migrate to a new
        device, so after a backup is restored to a new device these items will
        be missing.
    @constant kSecAttrAccessibleAlwaysThisDeviceOnly Item data can always
        be accessed regardless of the lock state of the device.  This option
        is not recommended for anything except system use. Items with this
        attribute will never migrate to a new device, so after a backup is
        restored to a new device, these items will be missing.
        
By default `kSecAttrAccessibleWhenUnlocked` - this constant is a cause of our issue. We can't write to Keychain if this constant.
As you can see `kSecAttrAccessibleAfterFirstUnlock` - This is recommended for items that need to be accesible by background applications. This our case. 
`kSecAttrAccessibleAlways` - **This is not recommended for anything except system use.** Items with this attribute will migrate to a new device when using encrypted backups.

I've tested the code and I haven't issue any more when Phone is locked and in background.